sdk/python/lib/pulumi/runtime/resource.py

```diff
@@ -123,7 +123,7 @@ class ResourceResolverOperations(NamedTuple):
     # former has been processed in the Resource constructor prior to calling
     # `register_resource` - both adding new inherited aliases and simplifying aliases down
     # to URNs.
-    aliases = []
+    aliases: List[Optional[str]] = []
```

Comment:
this certainly feels odd... i would not expect the aliases list to ever contain a None value.  i.e. i would expect `List[str]` with code filtering out anything that shouldn't be in there.